pkg/autodiscovery/integration/config_test.go

```diff
@@ -121,6 +121,15 @@ func TestDigest(t *testing.T) {
 	assert.Equal(t, "6253da85b1624771", simpleConfigWithLogs.Digest())
 }
 
+func TestGetExtraIDForInstance(t *testing.T) {
+	config := &Config{}
+
+	config.Name = "foo"
+	config.InitConfig = Data("fooBarBaz")
+	config.Instances = []Data{Data("namespace: foobar")}
+	assert.Equal(t, config.Instances[0].GetExtraIDForInstance(), "foobar")
```

Comment:
Could you please add a test with an instance with no `namespace` field?
This allows to make sure the logic does not `panic` on missing input